File: client/network.py
import socket

class Network:
    def __init__(self):
        self.client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        #self.server = "127.0.0.1"
        #self.server = "26.213.115.163"
        self.server = "82.179.140.18"
        self.port = 45295
        self.addr = (self.server, self.port)
        self.p = self.connect()

    def getP(self):
        return self.p

    def connect(self):
        try:
            self.client.connect(self.addr)
            return self.client.recv(1024)
        except:
            pass

    def wait(self):
        return self.client.recv(1024)
    
    def only_send(self, data):
        self.client.send(data)

    def send(self, data):
        try:
            self.client.send(data)
            return self.client.recv(1024)
        except socket.error as e:
            print(e)

File: Chess/Network.cs
using System.Net.Sockets;
using System.Net;
using System.Text;
using System.Windows.Documents;
using System;
using System.Windows;

namespace Chess
{
    internal class Network
    {
        Socket client = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
        int port = 45245;
        IPAddress adress = IPAddress.Parse("82.179.140.18");
        int BUFFLEN = 512;        
        public string con()
        {
            try
            {
                client.Connect(adress, port);
                // получаем данные
                byte[] responseBytes = new byte[BUFFLEN];
                int bytes = client.Receive(responseBytes);
                string response = Encoding.UTF8.GetString(responseBytes, 0, bytes);
                return response;
            }
            catch (Exception)
            {
                MessageBox.Show("Не удается подключиться к серверу", "Error", MessageBoxButton.OK, MessageBoxImage.Error, MessageBoxResult.Yes);
                Application.Current.Shutdown();
                return "error";
            }
            
        }
        public void snd(string str)
        {
            var sendBytes=Encoding.UTF8.GetBytes(str);
            client.Send(sendBytes);
        }
        public string rcv()
        {
            byte[] responseBytes = new byte[BUFFLEN];
            int bytes = client.Receive(responseBytes);
            string response = Encoding.UTF8.GetString(responseBytes, 0, bytes);
            return response;
        }
    }
}
